holoviews/core/data/__init__.py

```diff
@@ -527,14 +527,15 @@ def groupby(self, dimensions=[], container_type=HoloMap, group_type=None,
             kdims = [self.get_dimension(d) for d in group_dims]
             group_kwargs = dict(util.get_param_values(self), kdims=kdims)
             group_kwargs.update(kwargs)
+            kdims = group_kwargs['kdims']
             drop_dim = len(kdims) != len(group_kwargs['kdims'])
             def load_subset(*args):
                 constraint = dict(zip(dim_names, args))
                 group = self.select(**constraint)
                 if np.isscalar(group):
                     return group_type(([group],), group=self.group,
                                       label=self.label, vdims=self.vdims)
-                data = group.reindex(group_dims)
+                data = group.reindex(kdims)
```

Comment:
``kdims`` is shadowed a number of times. If ``len(kdims)`` is the same for line 527 and 530, I would inline the access on ``group_kwargs`` (which doesn't seem to be mutated anywhere). i.e:

``data = group.reindex(group_kwargs['kdims'])``

If the lengths are different, I am happy to discard this suggestion.

Reply:
Yah, I can probably clean this up a bit.

Reply:
Cleaned it up differently.